artifactory/commands/container/containermanagerbase.go

```diff
@@ -0,0 +1,100 @@
+package container
+
+import (
+	"github.com/jfrog/gofrog/version"
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/utils/container"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+)
+
+const (
+	RtMinVersion = "7.33.3"
+)
+
+type ContainerCommandBase struct {
+	image              *container.Image
+	repo               string
+	buildConfiguration *utils.BuildConfiguration
+	serverDetails      *config.ServerDetails
+}
+
+func (ccb *ContainerCommandBase) ImageTag() string {
+	return ccb.image.Name()
+}
+
+func (ccb *ContainerCommandBase) SetImageTag(imageTag string) *ContainerCommandBase {
+	ccb.image = container.NewImage(imageTag)
+	return ccb
+}
+
+func (ccb *ContainerCommandBase) Repo() (string, error) {
+	if ccb.repo != "" {
+		return ccb.repo, nil
+	}
+
+	serviceManager, err := utils.CreateServiceManager(ccb.serverDetails, -1, 0, false)
+	if err != nil {
+		return "", err
+	}
+	ccb.repo, err = ccb.image.GetRemoteRepo(serviceManager)
+	return ccb.repo, err
+}
+
+func (ccb *ContainerCommandBase) SetRepo(repo string) *ContainerCommandBase {
+	ccb.repo = repo
+	return ccb
+}
+
+// Since 'RtMinVersion' version of Artifactory we can fetch the docker repository without the user input (witch is deprecated).
+func (ccb *ContainerCommandBase) IsGetRepoSupported() (bool, error) {
+	serviceManager, err := utils.CreateServiceManager(ccb.serverDetails, -1, 0, false)
+	if err != nil {
+		return false, err
+	}
+	currentVersion, err := serviceManager.GetVersion()
+	if err != nil {
+		return false, err
+	}
+	if !version.NewVersion(currentVersion).AtLeast(RtMinVersion) {
+		return false, nil
+	}
+	return true, nil
+}
+
+func (ccb *ContainerCommandBase) BuildConfiguration() *utils.BuildConfiguration {
+	return ccb.buildConfiguration
+}
+
+func (ccb *ContainerCommandBase) SetBuildConfiguration(buildConfiguration *utils.BuildConfiguration) *ContainerCommandBase {
+	ccb.buildConfiguration = buildConfiguration
+	return ccb
+}
+
+func (ccb *ContainerCommandBase) ServerDetails() *config.ServerDetails {
+	return ccb.serverDetails
+}
+
+func (ccb *ContainerCommandBase) SetServerDetails(serverDetails *config.ServerDetails) *ContainerCommandBase {
+	ccb.serverDetails = serverDetails
+	return ccb
+}
+
+func (ccb *ContainerCommandBase) init() error {
+	toCollect, err := ccb.buildConfiguration.IsCollectBuildInfo()
+	if err != nil || !toCollect {
+		return err
+	}
+	if ccb.repo != "" {
+		return nil
+	}
+	// Check we have all we need to collect build-info.
+	ok, err := ccb.IsGetRepoSupported()
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return errorutils.CheckErrorf("Collecting build-info requires Artifactory version " + RtMinVersion + " or higher")
```

Comment:
Collecting build-info requires Artifactory version...
-->
Collecting docker build-info with this command requires Artifactory version...

Notice that the ```errorutils.CheckErrorf``` function supports string formats with ```%s```, which mases the usage more esthetic then using + between strings.